### PR TITLE
Add guide: "Easy profiling for Node.js"

### PR DESCRIPTION
**Pull request by matthewloring**
Adding a guide on the use of the tick processor and the built in Node.js
profiler to understand application performance.

### CODE REVIEW

**Comment by matthewloring** [comment]
Submitting this here on the recommendation of nodejs/docs#45

cc @Qard

**Comment by Qard** [comment]
LGTM. :+1:

**Comment by narqo** [comment]
It'd be very handy. :+1:

**Comment by evanlucas** [comment]
This looks awesome. LGTM

File: locale/en/docs/guides/simple-profiling.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+ 
+  if (users[username].hash.toString() === hash.toString()) {
+    res.sendStatus(200);
+  } else {
+    res.sendStatus(401);
+  }
+});
+```
+ 
+Please note that these are NOT recommended handlers for authenticating users in your Node.js applications and are used purely for illustration purposes. You should not be trying to design your own cryptographic authentication mechanisms in general. It is much better to use existing, proven authentication solutions.
```
Comment (stevemao) [inline locale/en/docs/guides/simple-profiling.md]: _Italic this sentence?_

File: locale/en/docs/guides/simple-profiling.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  75%   3818
+  80%   3825
+  90%   3845
+  95%   3858
+  98%   3874
+  99%   3875
+ 100%   4225 (longest request)
+```
+ 
+From this output, we see that we're only managing to serve about 5 requests per second and that the average request takes just under 4 seconds round trip. In a real world example, we could be doing lots of work in many functions on behalf of a user request but even in our simple example, time could be lost compiling regular expressions, generating random salts, generating  unique hashes from user passwords, or inside the Express framework itself.
```
Comment (stevemao) [inline locale/en/docs/guides/simple-profiling.md]: `generating  unique` contains an extra whitespace

Comment (matthewloring) [inline locale/en/docs/guides/simple-profiling.md]: Fixed.

File: locale/en/docs/guides/simple-profiling.md (hunk 1 above)
Comment (matthewloring) [inline locale/en/docs/guides/simple-profiling.md]: Sounds good!

**Comment by stevemao** [comment]
@nodejs/website  @nodejs/benchmarking

**Comment by phillipj** [comment]
LGTM

**Comment by fhemberger** [comment]
Merged, although it's not linked on the site yet.

**Comment by matthewloring** [comment]
Thanks! What is the process for getting this linked from the website?

**Comment by fhemberger** [comment]
@matthewloring Add a link to the website. ;)

**Comment by bnb** [comment]
@fhemberger Where would the link be added? I can probably handle it if nobody else has the time/desire.

**Comment by fhemberger** [comment]
@bnb Oh, that would be great! Create a new page "Guides" under "Docs" and add the link there.

Thanks!

**Comment by Qard** [comment]
The plan is to have a directory page for guides, topics and reference pages. I just haven't got around to making it yet. A static page would probably be fine for now too, if you want to push that through.

**Comment by matthewloring** [comment]
@bnb Were you able to look into including this in the website? I am happy to look into it but don't have any experience with the website code base yet.

**Comment by bnb** [comment]
@matthewloring Sorry, I've had a lot on my plate the last couple weeks (family member had heart surgery). I can try to do this now.

**Comment by bnb** [comment]
@fhemberger When you say create a new page "Guide" under "Docs", what _exactly_ do you mean? Sorry for being thick.

**Comment by fhemberger** [comment]
@bnb Add the page to the navigation structure in `site.json` and to `/layouts/docs.hbs`.

**Comment by bnb** [comment]
@fhemberger Thank you! :100:

**Pull request merged**